{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "3f5a5fe6",
   "metadata": {},
   "source": [
    "# C9orf72 age at onset analysis\n",
    "This notebook contains the source code to perform the age at onset analysis of the c9orf72 manuscript. The source data and analysis exist in the original folders but I want to create a folder to keep the essential data and code for Github and future references."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e37d7132",
   "metadata": {},
   "source": [
    "## Individual variant age at onset regression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0560762b",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[+] Loading gcc  9.2.0  ... \n",
      "[+] Loading GSL 2.6 for GCC 9.2.0 ...\n",
      "[-] Unloading gcc  9.2.0  ... \n",
      "[+] Loading gcc  9.2.0  ... \n",
      "[+] Loading openmpi 3.1.4  for GCC 9.2.0 \n",
      "[+] Loading ImageMagick  7.0.8  on cn1030 \n",
      "[+] Loading HDF5  1.10.4 \n",
      "[-] Unloading gcc  9.2.0  ... \n",
      "[+] Loading gcc  9.2.0  ... \n",
      "[+] Loading NetCDF 4.7.4_gcc9.2.0 \n",
      "[+] Loading pandoc  2.16.2  on cn1030 \n",
      "[+] Loading pcre2 10.21  ... \n",
      "[+] Loading R 4.0.0 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 4.0.0 (2020-04-24) -- \"Arbor Day\"\n",
      "Copyright (C) 2020 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "  Natural language support but running in an English locale\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> require(dplyr)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: dplyr\n",
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "Warning message:\n",
      "package ‘dplyr’ was built under R version 4.0.3 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> require(tidyverse)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: tidyverse\n",
      "── Attaching packages ─────────────────────────────────────── tidyverse 1.3.0 ──\n",
      "✔ ggplot2 3.3.5     ✔ purrr   0.3.4\n",
      "✔ tibble  3.1.6     ✔ stringr 1.4.0\n",
      "✔ tidyr   1.1.4     ✔ forcats 0.5.0\n",
      "✔ readr   2.1.1     \n",
      "── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "✖ dplyr::filter() masks stats::filter()\n",
      "✖ dplyr::lag()    masks stats::lag()\n",
      "Warning messages:\n",
      "1: package ‘ggplot2’ was built under R version 4.0.3 \n",
      "2: package ‘tibble’ was built under R version 4.0.3 \n",
      "3: package ‘tidyr’ was built under R version 4.0.3 \n",
      "4: package ‘readr’ was built under R version 4.0.3 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> require(ggplot2)\n",
      "> require(data.table)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: data.table\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "Warning message:\n",
      "package ‘data.table’ was built under R version 4.0.3 \n",
      "Loading required package: RColorBrewer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> require(RColorBrewer)\n",
      "> library(caTools)\n",
      "> \n",
      "> \n",
      "> profile= read.table(\"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2020-12-2.GRS/Results/Merged.162.RISK.profile\", header = T)\n",
      "> dim(profile)\n",
      "[1] 47184     6\n",
      "> profile$FID_IID = profile$FID\n",
      "> #PHENO FILE \n",
      "> pheno = read.table(\"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2021-03-26.MergingDatasets/PCs/UPDATED.COV.ALSplusALSFTD.C9.txt\", header = T)\n",
      "> dim(pheno)\n",
      "[1] 42090    38\n",
      "> \n",
      "> \n",
      "> #merge pheno with profile\n",
      "> data = merge(profile,pheno, by= \"FID_IID\")\n",
      "> \n",
      "> \n",
      "> table(data$diagnosis3)\n",
      "\n",
      "   Case Control \n",
      "   7854   34236 \n",
      "> table(data$diagnosis2)\n",
      "\n",
      "   Case_EXP     Case_WT Control_EXP  Control_WT \n",
      "        817        7037           1       34235 \n",
      "> #Case_EXP, Case_WT, Control_EXP, Control_WT\n",
      "> table(data$diagnosis4)\n",
      "\n",
      "    ALS Control     FTD \n",
      "   7703   34236     151 \n",
      "> \n",
      "> dim(data)\n",
      "[1] 42090    44\n",
      "> data$PHENO = data$PHENO.x\n",
      "> \n",
      "> # REGRESSION C9ORF72CARRIERS using GRS without C9orf72\n",
      "> data <- data %>% mutate (diagnosis5 = ifelse(diagnosis2 == \"Case_EXP\", \"Case_EXP\", \n",
      "+                                                         ifelse(diagnosis2 == \"Case_WT\", \"Case_WT\",\n",
      "+                                                                ifelse(diagnosis3 == \"Control\", \"Control\",\"no\"))))\n",
      "> \n",
      "> \n",
      "> data3groups = filter(data, diagnosis5 == \"Case_EXP\" |diagnosis5 == \"Case_WT\" | diagnosis5 == \"Control\")\n",
      "> dim(data3groups)\n",
      "[1] 42090    46\n",
      "> \n",
      "> data3groups = filter(data3groups, age_at_onset > 1)\n",
      "> dim(data3groups)\n",
      "[1] 42090    46\n",
      "> \n",
      "> # Whole dataset\n",
      "> \n",
      "> # open raw data file\n",
      "> raw = fread(\"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2020-12-2.GRS/Bin/162variants.Merged.Discovery.raw\", header = T)\n",
      "> # open raw data file\n",
      "> data3groups$FID = data3groups$FID.x\n",
      "> \n",
      "> data = merge(raw, data3groups, by = \"FID\")\n",
      "> \n",
      "> \n",
      "> \n",
      "> summary = data %>% group_by(diagnosis2) %>% \n",
      "+   summarize(mean_AAO = mean(age_at_onset), \n",
      "+             SD_AAO = sd(age_at_onset), count = n())\n",
      "> summary = as.data.table(summary)\n",
      "> print(summary)\n",
      "    diagnosis2 mean_AAO    SD_AAO count\n",
      "1:    Case_EXP 57.64382  9.587404   817\n",
      "2:     Case_WT 60.07645 12.522265  7037\n",
      "3: Control_EXP 82.00000        NA     1\n",
      "4:  Control_WT 63.40181 13.780152 34235\n",
      "> \n",
      "> \n",
      "> \n",
      "> ######## Additive model (effect of each additional allele) ###################\n",
      "> \n",
      "> dataexp = filter(data, diagnosis2 == \"Case_EXP\")\n",
      "> dim(dataexp)\n",
      "[1] 817 213\n",
      "> table(dataexp$dataset)\n",
      "\n",
      "ALS FTD \n",
      "666 151 \n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> AAO.merged.C9 <- lm(age_at_onset ~ rs9901522_T  + GENDER  + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = dataexp)\n",
      "> summary(AAO.merged.C9)\n",
      "\n",
      "Call:\n",
      "lm(formula = age_at_onset ~ rs9901522_T + GENDER + PC1 + PC2 + \n",
      "    PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + \n",
      "    PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = dataexp)\n",
      "\n",
      "Residuals:\n",
      "     Min       1Q   Median       3Q      Max \n",
      "-30.8405  -5.9748  -0.2085   6.5801  24.6449 \n",
      "\n",
      "Coefficients:\n",
      "            Estimate Std. Error t value Pr(>|t|)    \n",
      "(Intercept)  57.4391     0.5065 113.413  < 2e-16 ***\n",
      "rs9901522_T  -3.5127     0.9890  -3.552 0.000405 ***\n",
      "GENDER        0.4214     0.6740   0.625 0.532000    \n",
      "PC1          22.3056     6.5225   3.420 0.000659 ***\n",
      "PC2          -0.2716    16.1014  -0.017 0.986547    \n",
      "PC3         -34.7396    19.5576  -1.776 0.076071 .  \n",
      "PC4          -0.9093    15.2116  -0.060 0.952348    \n",
      "PC5          22.4974    26.1670   0.860 0.390178    \n",
      "PC6           9.2601    23.0995   0.401 0.688618    \n",
      "PC7           6.5430    26.4553   0.247 0.804724    \n",
      "PC8          18.2108    27.3861   0.665 0.506266    \n",
      "PC9         -34.8311    26.1937  -1.330 0.183982    \n",
      "PC10         18.6015    26.2854   0.708 0.479355    \n",
      "PC11         28.2095    26.8167   1.052 0.293148    \n",
      "PC12        -35.7252    26.6760  -1.339 0.180879    \n",
      "PC13         -0.1202    27.5684  -0.004 0.996522    \n",
      "PC14        -53.7140    27.5616  -1.949 0.051663 .  \n",
      "PC15        -28.7339    27.9758  -1.027 0.304687    \n",
      "PC16         -1.5485    27.3822  -0.057 0.954917    \n",
      "PC17          6.0598    27.8653   0.217 0.827901    \n",
      "PC18         36.0809    26.4623   1.363 0.173117    \n",
      "PC19        -22.8245    26.2166  -0.871 0.384228    \n",
      "PC20        -41.2652    27.2156  -1.516 0.129858    \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "Residual standard error: 9.467 on 794 degrees of freedom\n",
      "Multiple R-squared:  0.05115,\tAdjusted R-squared:  0.02486 \n",
      "F-statistic: 1.946 on 22 and 794 DF,  p-value: 0.005896\n",
      "\n",
      "> summary(AAO.merged.C9)$adj.r.squared\n",
      "[1] 0.02485865\n",
      "> \n",
      "> \n",
      "> AAO.merged.C9 <- lm(age_at_onset ~ rs113247976_T  + GENDER + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = dataexp)\n",
      "> summary(AAO.merged.C9)\n",
      "\n",
      "Call:\n",
      "lm(formula = age_at_onset ~ rs113247976_T + GENDER + PC1 + PC2 + \n",
      "    PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + \n",
      "    PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = dataexp)\n",
      "\n",
      "Residuals:\n",
      "     Min       1Q   Median       3Q      Max \n",
      "-30.2711  -6.2840  -0.1421   6.4958  25.1331 \n",
      "\n",
      "Coefficients:\n",
      "              Estimate Std. Error t value Pr(>|t|)    \n",
      "(Intercept)    57.2678     0.5066 113.052   <2e-16 ***\n",
      "rs113247976_T  -3.3597     1.6171  -2.078   0.0381 *  \n",
      "GENDER          0.3015     0.6772   0.445   0.6563    \n",
      "PC1            20.1381     6.5561   3.072   0.0022 ** \n",
      "PC2             1.6591    16.2014   0.102   0.9185    \n",
      "PC3           -35.9986    19.6607  -1.831   0.0675 .  \n",
      "PC4            -1.2001    15.2931  -0.078   0.9375    \n",
      "PC5            17.3635    26.3785   0.658   0.5106    \n",
      "PC6             5.5850    23.1980   0.241   0.8098    \n",
      "PC7             2.8039    26.6028   0.105   0.9161    \n",
      "PC8            16.3206    27.5264   0.593   0.5534    \n",
      "PC9           -38.0157    26.3318  -1.444   0.1492    \n",
      "PC10           18.1716    26.4381   0.687   0.4921    \n",
      "PC11           29.7072    26.9507   1.102   0.2707    \n",
      "PC12          -33.1283    26.8038  -1.236   0.2168    \n",
      "PC13            6.6044    27.6759   0.239   0.8115    \n",
      "PC14          -55.5713    27.7277  -2.004   0.0454 *  \n",
      "PC15          -26.6024    28.1154  -0.946   0.3443    \n",
      "PC16           -4.1292    27.5164  -0.150   0.8808    \n",
      "PC17            8.2533    27.9995   0.295   0.7683    \n",
      "PC18           36.4489    26.6013   1.370   0.1710    \n",
      "PC19          -24.6837    26.3447  -0.937   0.3491    \n",
      "PC20          -44.4559    27.3383  -1.626   0.1043    \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "Residual standard error: 9.517 on 794 degrees of freedom\n",
      "Multiple R-squared:  0.04129,\tAdjusted R-squared:  0.01472 \n",
      "F-statistic: 1.554 on 22 and 794 DF,  p-value: 0.05033\n",
      "\n",
      "> summary(AAO.merged.C9)$adj.r.squared\n",
      "[1] 0.01472275\n",
      "> \n",
      "> \n",
      "> \n",
      "> datnoaexp = filter(data, diagnosis2 == \"Case_WT\")\n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> AAO.nonC9 <- lm(age_at_onset ~ rs9901522_T   + GENDER  + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = datnoaexp)\n",
      "> summary(AAO.nonC9)\n",
      "\n",
      "Call:\n",
      "lm(formula = age_at_onset ~ rs9901522_T + GENDER + PC1 + PC2 + \n",
      "    PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + \n",
      "    PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = datnoaexp)\n",
      "\n",
      "Residuals:\n",
      "    Min      1Q  Median      3Q     Max \n",
      "-45.079  -7.896   1.067   8.889  47.709 \n",
      "\n",
      "Coefficients:\n",
      "            Estimate Std. Error t value Pr(>|t|)    \n",
      "(Intercept)  58.0552     0.2193 264.729  < 2e-16 ***\n",
      "rs9901522_T   0.1571     0.4297   0.365  0.71477    \n",
      "GENDER        3.0818     0.2991  10.302  < 2e-16 ***\n",
      "PC1          21.2484     2.8570   7.437 1.15e-13 ***\n",
      "PC2           9.7461     6.9310   1.406  0.15972    \n",
      "PC3          10.4451     8.4406   1.237  0.21595    \n",
      "PC4         -20.0924     7.0259  -2.860  0.00425 ** \n",
      "PC5         -21.4626    11.4297  -1.878  0.06045 .  \n",
      "PC6          45.0493    10.4156   4.325 1.55e-05 ***\n",
      "PC7         -17.8703    11.6147  -1.539  0.12395    \n",
      "PC8         -24.6349    11.6583  -2.113  0.03463 *  \n",
      "PC9          15.8280    11.8464   1.336  0.18156    \n",
      "PC10         -1.8195    11.9684  -0.152  0.87917    \n",
      "PC11         17.4140    11.7986   1.476  0.14001    \n",
      "PC12          4.2067    12.0078   0.350  0.72610    \n",
      "PC13          5.3467    11.9145   0.449  0.65362    \n",
      "PC14        -10.3039    11.7352  -0.878  0.37996    \n",
      "PC15        -10.9698    11.7596  -0.933  0.35093    \n",
      "PC16         -3.6284    11.6635  -0.311  0.75574    \n",
      "PC17         -4.2739    12.0184  -0.356  0.72214    \n",
      "PC18         15.9260    11.7887   1.351  0.17675    \n",
      "PC19          0.6865    12.0094   0.057  0.95442    \n",
      "PC20         -1.9010    11.8881  -0.160  0.87296    \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "Residual standard error: 12.35 on 7014 degrees of freedom\n",
      "Multiple R-squared:  0.03103,\tAdjusted R-squared:  0.02799 \n",
      "F-statistic: 10.21 on 22 and 7014 DF,  p-value: < 2.2e-16\n",
      "\n",
      "> summary(AAO.nonC9)$adj.r.squared\n",
      "[1] 0.02799428\n",
      "> \n",
      "> AAO.nonC9 <- lm(age_at_onset ~ rs113247976_T  + GENDER  + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = datnoaexp)\n",
      "> summary(AAO.nonC9)\n",
      "\n",
      "Call:\n",
      "lm(formula = age_at_onset ~ rs113247976_T + GENDER + PC1 + PC2 + \n",
      "    PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + \n",
      "    PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = datnoaexp)\n",
      "\n",
      "Residuals:\n",
      "    Min      1Q  Median      3Q     Max \n",
      "-45.097  -7.903   1.070   8.930  47.686 \n",
      "\n",
      "Coefficients:\n",
      "              Estimate Std. Error t value Pr(>|t|)    \n",
      "(Intercept)    58.0707     0.2140 271.422  < 2e-16 ***\n",
      "rs113247976_T   0.1456     0.8436   0.173  0.86295    \n",
      "GENDER          3.0811     0.2992  10.298  < 2e-16 ***\n",
      "PC1            21.2926     2.8580   7.450 1.04e-13 ***\n",
      "PC2             9.6777     6.9283   1.397  0.16251    \n",
      "PC3            10.4441     8.4412   1.237  0.21602    \n",
      "PC4           -20.1453     7.0260  -2.867  0.00415 ** \n",
      "PC5           -21.2945    11.4385  -1.862  0.06269 .  \n",
      "PC6            45.0085    10.4151   4.321 1.57e-05 ***\n",
      "PC7           -17.9025    11.6145  -1.541  0.12327    \n",
      "PC8           -24.6173    11.6586  -2.112  0.03476 *  \n",
      "PC9            15.8294    11.8465   1.336  0.18152    \n",
      "PC10           -1.8595    11.9679  -0.155  0.87653    \n",
      "PC11           17.5345    11.7979   1.486  0.13726    \n",
      "PC12            4.2231    12.0078   0.352  0.72508    \n",
      "PC13            5.5008    11.9146   0.462  0.64432    \n",
      "PC14          -10.2882    11.7354  -0.877  0.38069    \n",
      "PC15          -10.9855    11.7614  -0.934  0.35032    \n",
      "PC16           -3.6559    11.6640  -0.313  0.75396    \n",
      "PC17           -4.2501    12.0184  -0.354  0.72362    \n",
      "PC18           16.0601    11.7824   1.363  0.17291    \n",
      "PC19            0.7215    12.0108   0.060  0.95210    \n",
      "PC20           -1.9013    11.8889  -0.160  0.87295    \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "Residual standard error: 12.35 on 7014 degrees of freedom\n",
      "Multiple R-squared:  0.03102,\tAdjusted R-squared:  0.02798 \n",
      "F-statistic: 10.21 on 22 and 7014 DF,  p-value: < 2.2e-16\n",
      "\n",
      "> summary(AAO.nonC9)$adj.r.squared\n",
      "[1] 0.0279799\n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> ######## Dominant model (presence versus absence of minor allele) #############\n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> dataexp <- dataexp %>% mutate (rs9901522_T.dominant = ifelse(rs9901522_T == \"0\", \"0\", \n",
      "+                                                         ifelse(rs9901522_T == \"1\", \"1\",\n",
      "+                                                                ifelse(rs9901522_T ==  \"2\", \"1\",\"no\"))))\n",
      "> \n",
      "> \n",
      "> \n",
      "> dataexp <- dataexp %>% mutate (rs113247976_T.dominant = ifelse(rs113247976_T == \"0\", \"0\", \n",
      "+                                                         ifelse(rs113247976_T == \"1\", \"1\",\n",
      "+                                                                ifelse(rs113247976_T ==  \"2\", \"1\",\"no\"))))\n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> AAO.merged.C9 <- lm(age_at_onset ~ rs9901522_T.dominant  + GENDER + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = dataexp)\n",
      "> summary(AAO.merged.C9)\n",
      "\n",
      "Call:\n",
      "lm(formula = age_at_onset ~ rs9901522_T.dominant + GENDER + PC1 + \n",
      "    PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + \n",
      "    PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, \n",
      "    data = dataexp)\n",
      "\n",
      "Residuals:\n",
      "     Min       1Q   Median       3Q      Max \n",
      "-30.7537  -5.9880  -0.2986   6.5989  24.6304 \n",
      "\n",
      "Coefficients:\n",
      "                      Estimate Std. Error t value Pr(>|t|)    \n",
      "(Intercept)            57.4184     0.5071 113.225  < 2e-16 ***\n",
      "rs9901522_T.dominant1  -3.4875     1.0537  -3.310 0.000976 ***\n",
      "GENDER                  0.4191     0.6748   0.621 0.534669    \n",
      "PC1                    22.2652     6.5302   3.410 0.000683 ***\n",
      "PC2                    -0.4381    16.1184  -0.027 0.978323    \n",
      "PC3                   -34.9413    19.5774  -1.785 0.074679 .  \n",
      "PC4                    -1.3219    15.2253  -0.087 0.930836    \n",
      "PC5                    22.9906    26.1962   0.878 0.380410    \n",
      "PC6                     9.7024    23.1330   0.419 0.675023    \n",
      "PC7                     6.1175    26.4809   0.231 0.817363    \n",
      "PC8                    17.8381    27.4132   0.651 0.515421    \n",
      "PC9                   -35.6646    26.2173  -1.360 0.174106    \n",
      "PC10                   19.3612    26.3095   0.736 0.462007    \n",
      "PC11                   28.2676    26.8452   1.053 0.292668    \n",
      "PC12                  -35.7622    26.7053  -1.339 0.180908    \n",
      "PC13                    1.7038    27.5759   0.062 0.950748    \n",
      "PC14                  -53.2626    27.5896  -1.931 0.053897 .  \n",
      "PC15                  -27.6077    27.9944  -0.986 0.324342    \n",
      "PC16                   -1.8127    27.4096  -0.066 0.947289    \n",
      "PC17                    7.4554    27.8853   0.267 0.789261    \n",
      "PC18                   35.4639    26.4937   1.339 0.181091    \n",
      "PC19                  -21.7948    26.2588  -0.830 0.406788    \n",
      "PC20                  -41.4096    27.2448  -1.520 0.128931    \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "Residual standard error: 9.477 on 794 degrees of freedom\n",
      "Multiple R-squared:  0.04919,\tAdjusted R-squared:  0.02285 \n",
      "F-statistic: 1.867 on 22 and 794 DF,  p-value: 0.009309\n",
      "\n",
      "> summary(AAO.merged.C9)$adj.r.squared\n",
      "[1] 0.022848\n",
      "> \n",
      "> AAO.merged.C9 <- lm(age_at_onset ~ rs113247976_T.dominant  + GENDER  + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = dataexp)\n",
      "> summary(AAO.merged.C9)\n",
      "\n",
      "Call:\n",
      "lm(formula = age_at_onset ~ rs113247976_T.dominant + GENDER + \n",
      "    PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + \n",
      "    PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + \n",
      "    PC20, data = dataexp)\n",
      "\n",
      "Residuals:\n",
      "     Min       1Q   Median       3Q      Max \n",
      "-30.2711  -6.2840  -0.1421   6.4958  25.1331 \n",
      "\n",
      "Coefficients:\n",
      "                        Estimate Std. Error t value Pr(>|t|)    \n",
      "(Intercept)              57.2678     0.5066 113.052   <2e-16 ***\n",
      "rs113247976_T.dominant1  -3.3597     1.6171  -2.078   0.0381 *  \n",
      "GENDER                    0.3015     0.6772   0.445   0.6563    \n",
      "PC1                      20.1381     6.5561   3.072   0.0022 ** \n",
      "PC2                       1.6591    16.2014   0.102   0.9185    \n",
      "PC3                     -35.9986    19.6607  -1.831   0.0675 .  \n",
      "PC4                      -1.2001    15.2931  -0.078   0.9375    \n",
      "PC5                      17.3635    26.3785   0.658   0.5106    \n",
      "PC6                       5.5850    23.1980   0.241   0.8098    \n",
      "PC7                       2.8039    26.6028   0.105   0.9161    \n",
      "PC8                      16.3206    27.5264   0.593   0.5534    \n",
      "PC9                     -38.0157    26.3318  -1.444   0.1492    \n",
      "PC10                     18.1716    26.4381   0.687   0.4921    \n",
      "PC11                     29.7072    26.9507   1.102   0.2707    \n",
      "PC12                    -33.1283    26.8038  -1.236   0.2168    \n",
      "PC13                      6.6044    27.6759   0.239   0.8115    \n",
      "PC14                    -55.5713    27.7277  -2.004   0.0454 *  \n",
      "PC15                    -26.6024    28.1154  -0.946   0.3443    \n",
      "PC16                     -4.1292    27.5164  -0.150   0.8808    \n",
      "PC17                      8.2533    27.9995   0.295   0.7683    \n",
      "PC18                     36.4489    26.6013   1.370   0.1710    \n",
      "PC19                    -24.6837    26.3447  -0.937   0.3491    \n",
      "PC20                    -44.4559    27.3383  -1.626   0.1043    \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "Residual standard error: 9.517 on 794 degrees of freedom\n",
      "Multiple R-squared:  0.04129,\tAdjusted R-squared:  0.01472 \n",
      "F-statistic: 1.554 on 22 and 794 DF,  p-value: 0.05033\n",
      "\n",
      "> summary(AAO.merged.C9)$adj.r.squared\n",
      "[1] 0.01472275\n",
      "> \n",
      "> \n",
      "> \n",
      "> datnoaexp = filter(data, diagnosis2 == \"Case_WT\")\n",
      "> \n",
      "> \n",
      "> datnoaexp <- datnoaexp %>% mutate (rs9901522_T.dominant = ifelse(rs9901522_T == \"0\", \"0\", \n",
      "+                                                         ifelse(rs9901522_T == \"1\", \"1\",\n",
      "+                                                                ifelse(rs9901522_T ==  \"2\", \"1\",\"no\"))))\n",
      "> \n",
      "> \n",
      "> \n",
      "> datnoaexp <- datnoaexp %>% mutate (rs113247976_T.dominant = ifelse(rs113247976_T == \"0\", \"0\", \n",
      "+                                                         ifelse(rs113247976_T == \"1\", \"1\",\n",
      "+                                                                ifelse(rs113247976_T ==  \"2\", \"1\",\"no\"))))\n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> AAO.merged.C9 <- lm(age_at_onset ~ rs9901522_T.dominant  + GENDER + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = datnoaexp)\n",
      "> summary(AAO.merged.C9)\n",
      "\n",
      "Call:\n",
      "lm(formula = age_at_onset ~ rs9901522_T.dominant + GENDER + PC1 + \n",
      "    PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + \n",
      "    PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, \n",
      "    data = datnoaexp)\n",
      "\n",
      "Residuals:\n",
      "    Min      1Q  Median      3Q     Max \n",
      "-45.076  -7.895   1.070   8.891  47.713 \n",
      "\n",
      "Coefficients:\n",
      "                      Estimate Std. Error t value Pr(>|t|)    \n",
      "(Intercept)            58.0523     0.2195 264.494  < 2e-16 ***\n",
      "rs9901522_T.dominant1   0.1845     0.4463   0.413  0.67926    \n",
      "GENDER                  3.0820     0.2991  10.303  < 2e-16 ***\n",
      "PC1                    21.2431     2.8572   7.435 1.17e-13 ***\n",
      "PC2                     9.7565     6.9310   1.408  0.15928    \n",
      "PC3                    10.4403     8.4405   1.237  0.21616    \n",
      "PC4                   -20.0907     7.0258  -2.860  0.00425 ** \n",
      "PC5                   -21.4892    11.4304  -1.880  0.06015 .  \n",
      "PC6                    45.0555    10.4156   4.326 1.54e-05 ***\n",
      "PC7                   -17.8591    11.6148  -1.538  0.12419    \n",
      "PC8                   -24.6323    11.6582  -2.113  0.03465 *  \n",
      "PC9                    15.8187    11.8464   1.335  0.18182    \n",
      "PC10                   -1.8193    11.9683  -0.152  0.87918    \n",
      "PC11                   17.4056    11.7984   1.475  0.14019    \n",
      "PC12                    4.2016    12.0078   0.350  0.72642    \n",
      "PC13                    5.3507    11.9135   0.449  0.65335    \n",
      "PC14                  -10.3145    11.7352  -0.879  0.37947    \n",
      "PC15                  -10.9568    11.7598  -0.932  0.35151    \n",
      "PC16                   -3.6263    11.6635  -0.311  0.75588    \n",
      "PC17                   -4.2816    12.0185  -0.356  0.72166    \n",
      "PC18                   15.9156    11.7880   1.350  0.17701    \n",
      "PC19                    0.6971    12.0094   0.058  0.95371    \n",
      "PC20                   -1.8950    11.8881  -0.159  0.87336    \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "Residual standard error: 12.35 on 7014 degrees of freedom\n",
      "Multiple R-squared:  0.03104,\tAdjusted R-squared:  0.028 \n",
      "F-statistic: 10.21 on 22 and 7014 DF,  p-value: < 2.2e-16\n",
      "\n",
      "> summary(AAO.merged.C9)$adj.r.squared\n",
      "[1] 0.02799946\n",
      "> \n",
      "> AAO.merged.C9 <- lm(age_at_onset ~ rs113247976_T.dominant  + GENDER  + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = datnoaexp)\n",
      "> summary(AAO.merged.C9)\n",
      "\n",
      "Call:\n",
      "lm(formula = age_at_onset ~ rs113247976_T.dominant + GENDER + \n",
      "    PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + \n",
      "    PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + \n",
      "    PC20, data = datnoaexp)\n",
      "\n",
      "Residuals:\n",
      "    Min      1Q  Median      3Q     Max \n",
      "-45.098  -7.904   1.069   8.929  47.685 \n",
      "\n",
      "Coefficients:\n",
      "                        Estimate Std. Error t value Pr(>|t|)    \n",
      "(Intercept)              58.0714     0.2140 271.354  < 2e-16 ***\n",
      "rs113247976_T.dominant1   0.1250     0.8677   0.144  0.88547    \n",
      "GENDER                    3.0814     0.2992  10.299  < 2e-16 ***\n",
      "PC1                      21.2897     2.8580   7.449 1.05e-13 ***\n",
      "PC2                       9.6772     6.9283   1.397  0.16253    \n",
      "PC3                      10.4397     8.4410   1.237  0.21621    \n",
      "PC4                     -20.1424     7.0260  -2.867  0.00416 ** \n",
      "PC5                     -21.3084    11.4386  -1.863  0.06253 .  \n",
      "PC6                      45.0098    10.4151   4.322 1.57e-05 ***\n",
      "PC7                     -17.9067    11.6144  -1.542  0.12318    \n",
      "PC8                     -24.6157    11.6588  -2.111  0.03478 *  \n",
      "PC9                      15.8256    11.8465   1.336  0.18163    \n",
      "PC10                     -1.8590    11.9680  -0.155  0.87656    \n",
      "PC11                     17.5277    11.7978   1.486  0.13741    \n",
      "PC12                      4.2301    12.0080   0.352  0.72464    \n",
      "PC13                      5.4946    11.9150   0.461  0.64471    \n",
      "PC14                    -10.2834    11.7357  -0.876  0.38092    \n",
      "PC15                    -10.9928    11.7614  -0.935  0.35000    \n",
      "PC16                     -3.6567    11.6643  -0.313  0.75392    \n",
      "PC17                     -4.2498    12.0184  -0.354  0.72364    \n",
      "PC18                     16.0600    11.7824   1.363  0.17291    \n",
      "PC19                      0.7143    12.0106   0.059  0.95258    \n",
      "PC20                     -1.9016    11.8893  -0.160  0.87293    \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "Residual standard error: 12.35 on 7014 degrees of freedom\n",
      "Multiple R-squared:  0.03102,\tAdjusted R-squared:  0.02798 \n",
      "F-statistic: 10.21 on 22 and 7014 DF,  p-value: < 2.2e-16\n",
      "\n",
      "> summary(AAO.merged.C9)$adj.r.squared\n",
      "[1] 0.02797864\n",
      "> \n",
      "> # recesive model: presence versus absence of 2 minor alleles\n",
      "> # for kif5a we dont have 2 minor alleles genotype and only 4 for the other one, so it is pointless to do it.\n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2020-12-2.GRS/Results/\n",
    "\n",
    "# I am  trying to identify the SNPs that contribute the most to GRS. I regressed the genotype againts age at onset for every and each snp in decile 10.\n",
    "# Additionally, I split my data in training and replication to asses reproducibility\n",
    "\n",
    "module load R/4.0.0\n",
    "R --vanilla --no-save\n",
    "require(dplyr)\n",
    "require(tidyverse)\n",
    "require(ggplot2)\n",
    "require(data.table)\n",
    "require(RColorBrewer)\n",
    "library(caTools)\n",
    "\n",
    "\n",
    "profile= read.table(\"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2020-12-2.GRS/Results/Merged.162.RISK.profile\", header = T)\n",
    "dim(profile)\n",
    "profile$FID_IID = profile$FID\n",
    "#PHENO FILE \n",
    "pheno = read.table(\"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2021-03-26.MergingDatasets/PCs/UPDATED.COV.ALSplusALSFTD.C9.txt\", header = T)\n",
    "dim(pheno)\n",
    "\n",
    "\n",
    "#merge pheno with profile\n",
    "data = merge(profile,pheno, by= \"FID_IID\")\n",
    "\n",
    "\n",
    "table(data$diagnosis3)\n",
    "table(data$diagnosis2)\n",
    "#Case_EXP, Case_WT, Control_EXP, Control_WT\n",
    "table(data$diagnosis4)\n",
    "\n",
    "dim(data)\n",
    "data$PHENO = data$PHENO.x\n",
    "\n",
    "# REGRESSION C9ORF72CARRIERS using GRS without C9orf72\n",
    "data <- data %>% mutate (diagnosis5 = ifelse(diagnosis2 == \"Case_EXP\", \"Case_EXP\", \n",
    "                                                        ifelse(diagnosis2 == \"Case_WT\", \"Case_WT\",\n",
    "                                                               ifelse(diagnosis3 == \"Control\", \"Control\",\"no\"))))\n",
    "\n",
    "\n",
    "data3groups = filter(data, diagnosis5 == \"Case_EXP\" |diagnosis5 == \"Case_WT\" | diagnosis5 == \"Control\")\n",
    "dim(data3groups)\n",
    "\n",
    "data3groups = filter(data3groups, age_at_onset > 1)\n",
    "dim(data3groups)\n",
    "\n",
    "# Whole dataset\n",
    "\n",
    "# open raw data file\n",
    "raw = fread(\"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2020-12-2.GRS/Bin/162variants.Merged.Discovery.raw\", header = T)\n",
    "# open raw data file\n",
    "data3groups$FID = data3groups$FID.x\n",
    "\n",
    "data = merge(raw, data3groups, by = \"FID\")\n",
    "\n",
    "\n",
    "\n",
    "summary = data %>% group_by(diagnosis2) %>% \n",
    "  summarize(mean_AAO = mean(age_at_onset), \n",
    "            SD_AAO = sd(age_at_onset), count = n())\n",
    "summary = as.data.table(summary)\n",
    "print(summary)\n",
    "\n",
    "\n",
    "\n",
    "######## Additive model (effect of each additional allele) ###################\n",
    "\n",
    "dataexp = filter(data, diagnosis2 == \"Case_EXP\")\n",
    "dim(dataexp)\n",
    "table(dataexp$dataset)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "AAO.merged.C9 <- lm(age_at_onset ~ rs9901522_T  + GENDER  + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = dataexp)\n",
    "summary(AAO.merged.C9)\n",
    "summary(AAO.merged.C9)$adj.r.squared\n",
    "\n",
    "\n",
    "AAO.merged.C9 <- lm(age_at_onset ~ rs113247976_T  + GENDER + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = dataexp)\n",
    "summary(AAO.merged.C9)\n",
    "summary(AAO.merged.C9)$adj.r.squared\n",
    "\n",
    "\n",
    "\n",
    "datnoaexp = filter(data, diagnosis2 == \"Case_WT\")\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "AAO.nonC9 <- lm(age_at_onset ~ rs9901522_T   + GENDER  + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = datnoaexp)\n",
    "summary(AAO.nonC9)\n",
    "summary(AAO.nonC9)$adj.r.squared\n",
    "\n",
    "AAO.nonC9 <- lm(age_at_onset ~ rs113247976_T  + GENDER  + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = datnoaexp)\n",
    "summary(AAO.nonC9)\n",
    "summary(AAO.nonC9)$adj.r.squared\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "######## Dominant model (presence versus absence of minor allele) #############\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "dataexp <- dataexp %>% mutate (rs9901522_T.dominant = ifelse(rs9901522_T == \"0\", \"0\", \n",
    "                                                        ifelse(rs9901522_T == \"1\", \"1\",\n",
    "                                                               ifelse(rs9901522_T ==  \"2\", \"1\",\"no\"))))\n",
    "\n",
    "\n",
    "\n",
    "dataexp <- dataexp %>% mutate (rs113247976_T.dominant = ifelse(rs113247976_T == \"0\", \"0\", \n",
    "                                                        ifelse(rs113247976_T == \"1\", \"1\",\n",
    "                                                               ifelse(rs113247976_T ==  \"2\", \"1\",\"no\"))))\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "AAO.merged.C9 <- lm(age_at_onset ~ rs9901522_T.dominant  + GENDER + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = dataexp)\n",
    "summary(AAO.merged.C9)\n",
    "summary(AAO.merged.C9)$adj.r.squared\n",
    "\n",
    "AAO.merged.C9 <- lm(age_at_onset ~ rs113247976_T.dominant  + GENDER  + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = dataexp)\n",
    "summary(AAO.merged.C9)\n",
    "summary(AAO.merged.C9)$adj.r.squared\n",
    "\n",
    "\n",
    "\n",
    "datnoaexp = filter(data, diagnosis2 == \"Case_WT\")\n",
    "\n",
    "\n",
    "datnoaexp <- datnoaexp %>% mutate (rs9901522_T.dominant = ifelse(rs9901522_T == \"0\", \"0\", \n",
    "                                                        ifelse(rs9901522_T == \"1\", \"1\",\n",
    "                                                               ifelse(rs9901522_T ==  \"2\", \"1\",\"no\"))))\n",
    "\n",
    "\n",
    "\n",
    "datnoaexp <- datnoaexp %>% mutate (rs113247976_T.dominant = ifelse(rs113247976_T == \"0\", \"0\", \n",
    "                                                        ifelse(rs113247976_T == \"1\", \"1\",\n",
    "                                                               ifelse(rs113247976_T ==  \"2\", \"1\",\"no\"))))\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "AAO.merged.C9 <- lm(age_at_onset ~ rs9901522_T.dominant  + GENDER + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = datnoaexp)\n",
    "summary(AAO.merged.C9)\n",
    "summary(AAO.merged.C9)$adj.r.squared\n",
    "\n",
    "AAO.merged.C9 <- lm(age_at_onset ~ rs113247976_T.dominant  + GENDER  + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, data = datnoaexp)\n",
    "summary(AAO.merged.C9)\n",
    "summary(AAO.merged.C9)$adj.r.squared\n",
    "\n",
    "# recesive model: presence versus absence of 2 minor alleles\n",
    "# for kif5a we dont have 2 minor alleles genotype and only 4 for the other one, so it is pointless to do it.\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "79ca6e12",
   "metadata": {},
   "source": [
    "### Individual variants age at onset plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2a2702b8",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[+] Loading gcc  9.2.0  ... \n",
      "[+] Loading GSL 2.6 for GCC 9.2.0 ...\n",
      "[-] Unloading gcc  9.2.0  ... \n",
      "[+] Loading gcc  9.2.0  ... \n",
      "[+] Loading openmpi 3.1.4  for GCC 9.2.0 \n",
      "[+] Loading ImageMagick  7.0.8  on cn0912 \n",
      "[+] Loading HDF5  1.10.4 \n",
      "[-] Unloading gcc  9.2.0  ... \n",
      "[+] Loading gcc  9.2.0  ... \n",
      "[+] Loading NetCDF 4.7.4_gcc9.2.0 \n",
      "[+] Loading pandoc  2.17.1.1  on cn0912 \n",
      "[+] Loading pcre2 10.21  ... \n",
      "[+] Loading R 4.0.0 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 4.0.0 (2020-04-24) -- \"Arbor Day\"\n",
      "Copyright (C) 2020 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "  Natural language support but running in an English locale\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> require(dplyr)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: dplyr\n",
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "Warning message:\n",
      "package ‘dplyr’ was built under R version 4.0.3 \n",
      "Loading required package: tidyverse\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> require(tidyverse)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "── Attaching packages ─────────────────────────────────────── tidyverse 1.3.0 ──\n",
      "✔ ggplot2 3.3.5     ✔ purrr   0.3.4\n",
      "✔ tibble  3.1.6     ✔ stringr 1.4.0\n",
      "✔ tidyr   1.2.0     ✔ forcats 0.5.0\n",
      "✔ readr   2.1.1     \n",
      "── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "✖ dplyr::filter() masks stats::filter()\n",
      "✖ dplyr::lag()    masks stats::lag()\n",
      "Warning messages:\n",
      "1: package ‘ggplot2’ was built under R version 4.0.3 \n",
      "2: package ‘tibble’ was built under R version 4.0.3 \n",
      "3: package ‘tidyr’ was built under R version 4.0.5 \n",
      "4: package ‘readr’ was built under R version 4.0.3 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> require(ggplot2)\n",
      "> require(data.table)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: data.table\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "Warning message:\n",
      "package ‘data.table’ was built under R version 4.0.3 \n",
      "Loading required package: RColorBrewer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> require(RColorBrewer)\n",
      "> setwd(\"Merged.toPLOT.LOCAL.2021-04-11.GRS\")\n",
      "> \n",
      "> \n",
      "> \n",
      "> ############################################# OPEN .PROFILE AND COVARIATES FILES ##################################################\n",
      "> \n",
      "> profile= read.table(\"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2020-12-2.GRS/Results/Merged.162.RISK.profile\", header = T)\n",
      "> profile$FID_IID = profile$FID\n",
      "> #PHENO FILE \n",
      "> pheno = read.table(\"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2021-03-26.MergingDatasets/PCs/UPDATED.COV.ALSplusALSFTD.C9.txt\", header = T)\n",
      "> dim(pheno)\n",
      "[1] 42090    38\n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> #merge pheno with profile\n",
      "> data = merge(profile,pheno, by= \"FID_IID\")\n",
      "> \n",
      "> \n",
      "> table(data$diagnosis3)\n",
      "\n",
      "   Case Control \n",
      "   7854   34236 \n",
      "> table(data$diagnosis2)\n",
      "\n",
      "   Case_EXP     Case_WT Control_EXP  Control_WT \n",
      "        817        7037           1       34235 \n",
      "> #Case_EXP, Case_WT, Control_EXP, Control_WT\n",
      "> table(data$diagnosis4)\n",
      "\n",
      "    ALS Control     FTD \n",
      "   7703   34236     151 \n",
      "> \n",
      "> dim(data)\n",
      "[1] 42090    44\n",
      "> data$PHENO = data$PHENO.x\n",
      "> \n",
      "> # REGRESSION C9ORF72CARRIERS using GRS without C9orf72\n",
      "> data <- data %>% mutate (diagnosis5 = ifelse(diagnosis2 == \"Case_EXP\", \"Case_EXP\", \n",
      "+                                                         ifelse(diagnosis2 == \"Case_WT\", \"Case_WT\",\n",
      "+                                                                ifelse(diagnosis3 == \"Control\", \"Control\",\"no\"))))\n",
      "> \n",
      "> \n",
      "> data3groups = filter(data, diagnosis5 == \"Case_EXP\" |diagnosis5 == \"Case_WT\" | diagnosis5 == \"Control\")\n",
      "> dim(data3groups)\n",
      "[1] 42090    46\n",
      "> \n",
      "> data3groups = filter(data3groups, age_at_onset > 1)\n",
      "> dim(data3groups)\n",
      "[1] 42090    46\n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> # open raw data file\n",
      "> raw = fread(\"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2020-12-2.GRS/Bin/162variants.Merged.Discovery.raw\", header = T)\n",
      "> # open raw data file\n",
      "> data3groups$FID = data3groups$FID.x\n",
      "> \n",
      "> data = merge(raw, data3groups, by = \"FID\")\n",
      "> \n",
      "> data = filter(data, diagnosis2 == \"Case_EXP\")\n",
      "> \n",
      "> \n",
      "> \n",
      "> data = filter(data, diagnosis2 == \"Case_EXP\")\n",
      "> table(data$diagnosis2)\n",
      "\n",
      "Case_EXP \n",
      "     817 \n",
      "> \n",
      "> p <- ggplot(data, aes(x = as.factor(rs9901522_T), y=age_at_onset, fill=rs9901522_T)) +\n",
      "+   geom_boxplot(width=0.6, fatten = 4, fill=c('#354E71', '#354E71', '#354E71'), alpha = 1/4,  colour = c('#354E71', '#354E71', '#354E71')) + \n",
      "+   theme_classic(base_size = 40) + coord_flip()\n",
      ">  \n",
      ">  p2 = p + ylab(\"Age at onset\") + xlab(\"rs9901522 Genotype\") +\n",
      "+ scale_x_discrete(name = \"rs9901522 Genotype\", breaks=c(\"0\",\"1\", \"2\"), labels=c(\"CC\",\"CT\", \"TT\")) + \n",
      "+ theme(plot.title = element_text(size = 40), legend.position = \"none\")\n",
      "> \n",
      "> #ggsave(filename = \"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/Merged.toPLOT.LOCAL.2021-04-11.GRS/updated.BoxPlot_rs9901522.Merged.png\", width = 40, height = 30, dpi = 300, units = \"cm\", plot = p2) \n",
      "> \n",
      "> \n",
      "> p <- ggplot(data, aes(x = as.factor(rs113247976_T), y=age_at_onset, fill=rs113247976_T)) +\n",
      "+   geom_boxplot(width=0.6, fatten = 4, fill=c('#CE4A7EFF', '#CE4A7EFF'), alpha = 1/4,  \n",
      "+   colour = c('#CE4A7EFF', '#CE4A7EFF')) + \n",
      "+   theme_classic(base_size = 40) + coord_flip()\n",
      ">   p2 = p + ylab(\"Age at onset\") + xlab(\"rs113247976_T Genotype\") +\n",
      "+ scale_x_discrete(name = \"rs113247976 Genotype\", breaks=c(\"0\",\"1\"), labels=c(\"CC\",\"CT\")) + \n",
      "+ theme(plot.title = element_text(size = 40), legend.position = \"none\")\n",
      "> \n",
      "> \n",
      "> #ggsave(filename = \"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/Merged.toPLOT.LOCAL.2021-04-11.GRS/updated.BoxPlot_rs113247976.Merged.png\", width = 40, height = 20, dpi = 300, units = \"cm\", plot = p2) \n",
      "> \n",
      "> \n",
      "> p <- ggplot(data, aes(x = diagnosis5, y=age_at_onset, fill=diagnosis5)) +\n",
      "+   geom_boxplot(width=0.6, fatten = 4, fill=c('grey24'), alpha = 1/4,  \n",
      "+   colour = c('grey24')) + \n",
      "+   theme_classic(base_size = 40) + coord_flip()\n",
      ">   p2 = p + ylab(\"Age at onset\") + xlab(\"C9orf72 cases\") +\n",
      "+ scale_x_discrete(name = \"C9orf72 cases\", breaks=c(\"Case_EXP\"), labels=c(\"\")) + \n",
      "+ theme(plot.title = element_text(size = 40), legend.position = \"none\")\n",
      "> \n",
      "> #ggsave(filename = \"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/Merged.toPLOT.LOCAL.2021-04-11.GRS/updated.BoxPlot_C9orf72cases.Merged.png\", width = 40, height = 20, dpi = 300, units = \"cm\", plot = p2) \n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> data = merge(raw, data3groups, by = \"FID\")\n",
      "> data = filter(data, diagnosis2 == \"Case_WT\")\n",
      "> \n",
      "> p <- ggplot(data, aes(x = as.factor(rs9901522_T), y=age_at_onset, fill=rs9901522_T)) +\n",
      "+   geom_boxplot(width=0.6, fatten = 4, fill=c('#354E71', '#354E71', '#354E71'), alpha = 1/4,  colour = c('#354E71', '#354E71', '#354E71')) + \n",
      "+   theme_classic(base_size = 40) + coord_flip()\n",
      ">   p2 = p + ylab(\"Age at onset\") + xlab(\"rs9901522 Genotype\") +\n",
      "+ scale_x_discrete(name = \"rs9901522 Genotype\", breaks=c(\"0\",\"1\", \"2\"), labels=c(\"CC\",\"CT\", \"TT\")) + \n",
      "+ theme(plot.title = element_text(size = 40), legend.position = \"none\")\n",
      "> \n",
      "> \n",
      "> #ggsave(filename = \"BoxPlot_rs9901522.Merged.WT.png\", width = 40, height = 30, dpi = 300, units = \"cm\", plot = p2) \n",
      "> \n",
      "> data = merge(raw, data3groups, by = \"FID\")\n",
      "> \n",
      "> data = filter(data, diagnosis2 == \"Case_WT\")\n",
      "> \n",
      "> \n",
      "> \n",
      "> p <- ggplot(data, aes(x = diagnosis5, y=age_at_onset, fill=diagnosis5)) +\n",
      "+   geom_boxplot(width=0.6, fatten = 4, fill=c('grey24'), alpha = 1/4,  \n",
      "+   colour = c('grey24')) + \n",
      "+   theme_classic(base_size = 40) + coord_flip()\n",
      ">   p2 = p + ylab(\"Age at onset\") + xlab(\"C9orf72 cases\") +\n",
      "+ scale_x_discrete(name = \"Non cases\", breaks=c(\"Case_WT\"), labels=c(\"\")) + \n",
      "+ theme(plot.title = element_text(size = 40), legend.position = \"none\")\n",
      "> \n",
      "> \n",
      "> #ggsave(filename = \"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/Merged.toPLOT.LOCAL.2021-04-11.GRS/2updated.BoxPlot_C9orf72cases.Merged.WT.png\", width = 40, height = 20, dpi = 300, units = \"cm\", plot = p2) \n",
      "> \n",
      "> data = filter (data, rs113247976_T == \"1\" |  rs113247976_T == \"0\")\n",
      "> \n",
      "> p <- ggplot(data, aes(x = as.factor(rs113247976_T), y=age_at_onset, fill= rs113247976_T)) +\n",
      "+   geom_boxplot(width=0.6, fatten = 4, fill=c('#CE4A7EFF', '#CE4A7EFF'), alpha = 1/4,  \n",
      "+   colour = c('#CE4A7EFF', '#CE4A7EFF')) + \n",
      "+   theme_classic(base_size = 40) + coord_flip()\n",
      ">   p2 = p + ylab(\"Age at onset\") + xlab(\"rs113247976 Genotype\") +\n",
      "+ scale_x_discrete(name = \"rs113247976 Genotype\", breaks=c(\"0\",\"1\"), labels=c(\"CC\",\"CT\")) + \n",
      "+ theme(plot.title = element_text(size = 40), legend.position = \"none\")\n",
      "> \n",
      "> #ggsave(filename = \"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/Merged.toPLOT.LOCAL.2021-04-11.GRS/updated.BoxPlot_rs113247976.Merged.WT.png\", width = 40, height = 20, dpi = 300, units = \"cm\", plot = p2) \n",
      "> \n",
      "> data = merge(raw, data3groups, by = \"FID\")\n",
      "> data = filter(data, diagnosis2 == \"Case_WT\")\n",
      "> \n",
      "> \n",
      "> \n",
      "> p <- ggplot(data, aes(x = as.factor(rs9901522_T), y=age_at_onset, fill=rs9901522_T)) +\n",
      "+   geom_boxplot(width=0.6, fatten = 4, fill=c('#354E71', '#354E71', '#354E71'), alpha = 1/4,  colour = c('#354E71', '#354E71', '#354E71')) + \n",
      "+   theme_classic(base_size = 40) + coord_flip()\n",
      ">   p2 = p + ylab(\"Age at onset\") + xlab(\"rs9901522 Genotype\") +\n",
      "+ scale_x_discrete(name = \"rs9901522 Genotype\", breaks=c(\"0\",\"1\", \"2\"), labels=c(\"CC\",\"CT\", \"TT\")) + \n",
      "+ theme(plot.title = element_text(size = 40), legend.position = \"none\")\n",
      "> \n",
      "> \n",
      "> #ggsave(filename = \"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/Merged.toPLOT.LOCAL.2021-04-11.GRS/BoxPlot_rs9901522.Merged.WT.png\", width = 40, height = 30, dpi = 300, units = \"cm\", plot = p2) \n",
      "> \n",
      "> profile= read.table(\"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2020-12-2.GRS/Results/Merged.162.RISK.profile\", header = T)\n",
      "> \n",
      "> profile$FID_IID = profile$FID\n",
      "> #PHENO FILE \n",
      "> pheno = read.table(\"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2021-03-26.MergingDatasets/PCs/UPDATED.COV.ALSplusALSFTD.C9.txt\", header = T)\n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> #merge pheno with profile\n",
      "> data = merge(profile,pheno, by= \"FID_IID\")\n",
      "> \n",
      "> \n",
      "> table(data$diagnosis3)\n",
      "\n",
      "   Case Control \n",
      "   7854   34236 \n",
      "> table(data$diagnosis2)\n",
      "\n",
      "   Case_EXP     Case_WT Control_EXP  Control_WT \n",
      "        817        7037           1       34235 \n",
      "> #Case_EXP, Case_WT, Control_EXP, Control_WT\n",
      "> table(data$diagnosis4)\n",
      "\n",
      "    ALS Control     FTD \n",
      "   7703   34236     151 \n",
      "> \n",
      "> dim(data)\n",
      "[1] 42090    44\n",
      "> data$PHENO = data$PHENO.x\n",
      "> \n",
      "> # REGRESSION C9ORF72CARRIERS using GRS without C9orf72\n",
      "> data <- data %>% mutate (diagnosis5 = ifelse(diagnosis2 == \"Case_EXP\", \"Case_EXP\", \n",
      "+                                                         ifelse(diagnosis2 == \"Case_WT\", \"Case_WT\",\n",
      "+                                                                ifelse(diagnosis3 == \"Control\", \"Control\",\"no\"))))\n",
      "> \n",
      "> \n",
      "> data3groups = filter(data, diagnosis5 == \"Case_EXP\" |diagnosis5 == \"Case_WT\" | diagnosis5 == \"Control\")\n",
      "> dim(data3groups)\n",
      "[1] 42090    46\n",
      "> \n",
      "> data3groups = filter(data3groups, age_at_onset > 1)\n",
      "> dim(data3groups)\n",
      "[1] 42090    46\n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> # open raw data file\n",
      "> raw = fread(\"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2020-12-2.GRS/Bin/162variants.Merged.Discovery.raw\", header = T)\n",
      "> # open raw data file\n",
      "> data3groups$FID = data3groups$FID.x\n",
      "> \n",
      "> data = merge(raw, data3groups, by = \"FID\")\n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> data1 = data %>% group_by(diagnosis2, rs113247976_T) %>% \n",
      "+   summarize(mean_AAO = mean(age_at_onset), \n",
      "+             SD_AAO = sd(age_at_onset))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "`summarise()` has grouped output by 'diagnosis2'. You can override using the `.groups` argument.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> data1 = as.data.table(data1)\n",
      "> print(data1)\n",
      "    diagnosis2 rs113247976_T mean_AAO    SD_AAO\n",
      "1:    Case_EXP             0 57.80769  9.487317\n",
      "2:    Case_EXP             1 54.18919 11.087625\n",
      "3:     Case_WT             0 60.07588 12.535801\n",
      "4:     Case_WT             1 60.06311 12.135049\n",
      "5:     Case_WT             2 61.75000 11.757976\n",
      "6: Control_EXP             0 82.00000        NA\n",
      "7:  Control_WT             0 63.39280 13.795042\n",
      "8:  Control_WT             1 63.77818 13.179143\n",
      "9:  Control_WT             2 61.83333 12.890565\n",
      "> \n",
      "> data2 = data %>% group_by(diagnosis2, rs9901522_T) %>% \n",
      "+   summarize(mean_AAO = mean(age_at_onset), \n",
      "+             SD_AAO = sd(age_at_onset))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "`summarise()` has grouped output by 'diagnosis2'. You can override using the `.groups` argument.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> data2 = as.data.table(data2)\n",
      "> print(data2)\n",
      "     diagnosis2 rs9901522_T mean_AAO    SD_AAO\n",
      " 1:    Case_EXP           0 58.02762  9.310195\n",
      " 2:    Case_EXP           1 54.96629 11.219352\n",
      " 3:    Case_EXP           2 47.75000  6.396614\n",
      " 4:     Case_WT           0 60.04416 12.527851\n",
      " 5:     Case_WT           1 60.31776 12.487447\n",
      " 6:     Case_WT           2 59.72727 12.780870\n",
      " 7: Control_EXP           0 82.00000        NA\n",
      " 8:  Control_WT           0 63.41266 13.752461\n",
      " 9:  Control_WT           1 63.29784 14.017007\n",
      "10:  Control_WT           2 63.90909 13.387183\n",
      "> \n",
      "> \n",
      ">     \n",
      "> data3 = data %>% group_by(diagnosis2) %>% \n",
      "+   summarize(mean_AAO = mean(age_at_onset), \n",
      "+             SD_AAO = sd(age_at_onset),\n",
      "+             n = n())\n",
      "> data3 = as.data.table(data3)\n",
      "> print(data3)\n",
      "    diagnosis2 mean_AAO    SD_AAO     n\n",
      "1:    Case_EXP 57.64382  9.587404   817\n",
      "2:     Case_WT 60.07645 12.522265  7037\n",
      "3: Control_EXP 82.00000        NA     1\n",
      "4:  Control_WT 63.40181 13.780152 34235\n",
      ">   \n",
      "> data3 = data %>% group_by(diagnosis3) %>% \n",
      "+   summarize(mean_AAO = mean(age_at_onset), \n",
      "+             SD_AAO = sd(age_at_onset),\n",
      "+             n = n())\n",
      "> data3 = as.data.table(data3)\n",
      "> print(data3)\n",
      "   diagnosis3 mean_AAO   SD_AAO     n\n",
      "1:       Case 59.82340 12.27176  7854\n",
      "2:    Control 63.40235 13.78032 34236\n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "\n",
    "module load R/4.0.0\n",
    "R --vanilla --no-save\n",
    "require(dplyr)\n",
    "require(tidyverse)\n",
    "require(ggplot2)\n",
    "require(data.table)\n",
    "require(RColorBrewer)\n",
    "setwd(\"Merged.toPLOT.LOCAL.2021-04-11.GRS\")\n",
    "\n",
    "\n",
    "\n",
    "############################################# OPEN .PROFILE AND COVARIATES FILES ##################################################\n",
    "\n",
    "profile= read.table(\"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2020-12-2.GRS/Results/Merged.162.RISK.profile\", header = T)\n",
    "profile$FID_IID = profile$FID\n",
    "#PHENO FILE \n",
    "pheno = read.table(\"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2021-03-26.MergingDatasets/PCs/UPDATED.COV.ALSplusALSFTD.C9.txt\", header = T)\n",
    "dim(pheno)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#merge pheno with profile\n",
    "data = merge(profile,pheno, by= \"FID_IID\")\n",
    "\n",
    "\n",
    "table(data$diagnosis3)\n",
    "table(data$diagnosis2)\n",
    "#Case_EXP, Case_WT, Control_EXP, Control_WT\n",
    "table(data$diagnosis4)\n",
    "\n",
    "dim(data)\n",
    "data$PHENO = data$PHENO.x\n",
    "\n",
    "# REGRESSION C9ORF72CARRIERS using GRS without C9orf72\n",
    "data <- data %>% mutate (diagnosis5 = ifelse(diagnosis2 == \"Case_EXP\", \"Case_EXP\", \n",
    "                                                        ifelse(diagnosis2 == \"Case_WT\", \"Case_WT\",\n",
    "                                                               ifelse(diagnosis3 == \"Control\", \"Control\",\"no\"))))\n",
    "\n",
    "\n",
    "data3groups = filter(data, diagnosis5 == \"Case_EXP\" |diagnosis5 == \"Case_WT\" | diagnosis5 == \"Control\")\n",
    "dim(data3groups)\n",
    "\n",
    "data3groups = filter(data3groups, age_at_onset > 1)\n",
    "dim(data3groups)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# open raw data file\n",
    "raw = fread(\"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2020-12-2.GRS/Bin/162variants.Merged.Discovery.raw\", header = T)\n",
    "# open raw data file\n",
    "data3groups$FID = data3groups$FID.x\n",
    "\n",
    "data = merge(raw, data3groups, by = \"FID\")\n",
    "\n",
    "data = filter(data, diagnosis2 == \"Case_EXP\")\n",
    "\n",
    "\n",
    "\n",
    "data = filter(data, diagnosis2 == \"Case_EXP\")\n",
    "table(data$diagnosis2)\n",
    "\n",
    "p <- ggplot(data, aes(x = as.factor(rs9901522_T), y=age_at_onset, fill=rs9901522_T)) +\n",
    "  geom_boxplot(width=0.6, fatten = 4, fill=c('#354E71', '#354E71', '#354E71'), alpha = 1/4,  colour = c('#354E71', '#354E71', '#354E71')) + \n",
    "  theme_classic(base_size = 40) + coord_flip()\n",
    " \n",
    " p2 = p + ylab(\"Age at onset\") + xlab(\"rs9901522 Genotype\") +\n",
    "scale_x_discrete(name = \"rs9901522 Genotype\", breaks=c(\"0\",\"1\", \"2\"), labels=c(\"CC\",\"CT\", \"TT\")) + \n",
    "theme(plot.title = element_text(size = 40), legend.position = \"none\")\n",
    "\n",
    "#ggsave(filename = \"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/Merged.toPLOT.LOCAL.2021-04-11.GRS/updated.BoxPlot_rs9901522.Merged.png\", width = 40, height = 30, dpi = 300, units = \"cm\", plot = p2) \n",
    "\n",
    "\n",
    "p <- ggplot(data, aes(x = as.factor(rs113247976_T), y=age_at_onset, fill=rs113247976_T)) +\n",
    "  geom_boxplot(width=0.6, fatten = 4, fill=c('#CE4A7EFF', '#CE4A7EFF'), alpha = 1/4,  \n",
    "  colour = c('#CE4A7EFF', '#CE4A7EFF')) + \n",
    "  theme_classic(base_size = 40) + coord_flip()\n",
    "  p2 = p + ylab(\"Age at onset\") + xlab(\"rs113247976_T Genotype\") +\n",
    "scale_x_discrete(name = \"rs113247976 Genotype\", breaks=c(\"0\",\"1\"), labels=c(\"CC\",\"CT\")) + \n",
    "theme(plot.title = element_text(size = 40), legend.position = \"none\")\n",
    "\n",
    "\n",
    "#ggsave(filename = \"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/Merged.toPLOT.LOCAL.2021-04-11.GRS/updated.BoxPlot_rs113247976.Merged.png\", width = 40, height = 20, dpi = 300, units = \"cm\", plot = p2) \n",
    "\n",
    "\n",
    "p <- ggplot(data, aes(x = diagnosis5, y=age_at_onset, fill=diagnosis5)) +\n",
    "  geom_boxplot(width=0.6, fatten = 4, fill=c('grey24'), alpha = 1/4,  \n",
    "  colour = c('grey24')) + \n",
    "  theme_classic(base_size = 40) + coord_flip()\n",
    "  p2 = p + ylab(\"Age at onset\") + xlab(\"C9orf72 cases\") +\n",
    "scale_x_discrete(name = \"C9orf72 cases\", breaks=c(\"Case_EXP\"), labels=c(\"\")) + \n",
    "theme(plot.title = element_text(size = 40), legend.position = \"none\")\n",
    "\n",
    "#ggsave(filename = \"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/Merged.toPLOT.LOCAL.2021-04-11.GRS/updated.BoxPlot_C9orf72cases.Merged.png\", width = 40, height = 20, dpi = 300, units = \"cm\", plot = p2) \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "data = merge(raw, data3groups, by = \"FID\")\n",
    "data = filter(data, diagnosis2 == \"Case_WT\")\n",
    "\n",
    "p <- ggplot(data, aes(x = as.factor(rs9901522_T), y=age_at_onset, fill=rs9901522_T)) +\n",
    "  geom_boxplot(width=0.6, fatten = 4, fill=c('#354E71', '#354E71', '#354E71'), alpha = 1/4,  colour = c('#354E71', '#354E71', '#354E71')) + \n",
    "  theme_classic(base_size = 40) + coord_flip()\n",
    "  p2 = p + ylab(\"Age at onset\") + xlab(\"rs9901522 Genotype\") +\n",
    "scale_x_discrete(name = \"rs9901522 Genotype\", breaks=c(\"0\",\"1\", \"2\"), labels=c(\"CC\",\"CT\", \"TT\")) + \n",
    "theme(plot.title = element_text(size = 40), legend.position = \"none\")\n",
    "\n",
    "\n",
    "#ggsave(filename = \"BoxPlot_rs9901522.Merged.WT.png\", width = 40, height = 30, dpi = 300, units = \"cm\", plot = p2) \n",
    "\n",
    "data = merge(raw, data3groups, by = \"FID\")\n",
    "\n",
    "data = filter(data, diagnosis2 == \"Case_WT\")\n",
    "\n",
    "\n",
    "\n",
    "p <- ggplot(data, aes(x = diagnosis5, y=age_at_onset, fill=diagnosis5)) +\n",
    "  geom_boxplot(width=0.6, fatten = 4, fill=c('grey24'), alpha = 1/4,  \n",
    "  colour = c('grey24')) + \n",
    "  theme_classic(base_size = 40) + coord_flip()\n",
    "  p2 = p + ylab(\"Age at onset\") + xlab(\"C9orf72 cases\") +\n",
    "scale_x_discrete(name = \"Non cases\", breaks=c(\"Case_WT\"), labels=c(\"\")) + \n",
    "theme(plot.title = element_text(size = 40), legend.position = \"none\")\n",
    "\n",
    "\n",
    "#ggsave(filename = \"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/Merged.toPLOT.LOCAL.2021-04-11.GRS/2updated.BoxPlot_C9orf72cases.Merged.WT.png\", width = 40, height = 20, dpi = 300, units = \"cm\", plot = p2) \n",
    "\n",
    "data = filter (data, rs113247976_T == \"1\" |  rs113247976_T == \"0\")\n",
    "\n",
    "p <- ggplot(data, aes(x = as.factor(rs113247976_T), y=age_at_onset, fill= rs113247976_T)) +\n",
    "  geom_boxplot(width=0.6, fatten = 4, fill=c('#CE4A7EFF', '#CE4A7EFF'), alpha = 1/4,  \n",
    "  colour = c('#CE4A7EFF', '#CE4A7EFF')) + \n",
    "  theme_classic(base_size = 40) + coord_flip()\n",
    "  p2 = p + ylab(\"Age at onset\") + xlab(\"rs113247976 Genotype\") +\n",
    "scale_x_discrete(name = \"rs113247976 Genotype\", breaks=c(\"0\",\"1\"), labels=c(\"CC\",\"CT\")) + \n",
    "theme(plot.title = element_text(size = 40), legend.position = \"none\")\n",
    "\n",
    "#ggsave(filename = \"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/Merged.toPLOT.LOCAL.2021-04-11.GRS/updated.BoxPlot_rs113247976.Merged.WT.png\", width = 40, height = 20, dpi = 300, units = \"cm\", plot = p2) \n",
    "\n",
    "data = merge(raw, data3groups, by = \"FID\")\n",
    "data = filter(data, diagnosis2 == \"Case_WT\")\n",
    "\n",
    "\n",
    "\n",
    "p <- ggplot(data, aes(x = as.factor(rs9901522_T), y=age_at_onset, fill=rs9901522_T)) +\n",
    "  geom_boxplot(width=0.6, fatten = 4, fill=c('#354E71', '#354E71', '#354E71'), alpha = 1/4,  colour = c('#354E71', '#354E71', '#354E71')) + \n",
    "  theme_classic(base_size = 40) + coord_flip()\n",
    "  p2 = p + ylab(\"Age at onset\") + xlab(\"rs9901522 Genotype\") +\n",
    "scale_x_discrete(name = \"rs9901522 Genotype\", breaks=c(\"0\",\"1\", \"2\"), labels=c(\"CC\",\"CT\", \"TT\")) + \n",
    "theme(plot.title = element_text(size = 40), legend.position = \"none\")\n",
    "\n",
    "\n",
    "#ggsave(filename = \"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/Merged.toPLOT.LOCAL.2021-04-11.GRS/BoxPlot_rs9901522.Merged.WT.png\", width = 40, height = 30, dpi = 300, units = \"cm\", plot = p2) \n",
    "\n",
    "profile= read.table(\"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2020-12-2.GRS/Results/Merged.162.RISK.profile\", header = T)\n",
    "\n",
    "profile$FID_IID = profile$FID\n",
    "#PHENO FILE \n",
    "pheno = read.table(\"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2021-03-26.MergingDatasets/PCs/UPDATED.COV.ALSplusALSFTD.C9.txt\", header = T)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#merge pheno with profile\n",
    "data = merge(profile,pheno, by= \"FID_IID\")\n",
    "\n",
    "\n",
    "table(data$diagnosis3)\n",
    "table(data$diagnosis2)\n",
    "#Case_EXP, Case_WT, Control_EXP, Control_WT\n",
    "table(data$diagnosis4)\n",
    "\n",
    "dim(data)\n",
    "data$PHENO = data$PHENO.x\n",
    "\n",
    "# REGRESSION C9ORF72CARRIERS using GRS without C9orf72\n",
    "data <- data %>% mutate (diagnosis5 = ifelse(diagnosis2 == \"Case_EXP\", \"Case_EXP\", \n",
    "                                                        ifelse(diagnosis2 == \"Case_WT\", \"Case_WT\",\n",
    "                                                               ifelse(diagnosis3 == \"Control\", \"Control\",\"no\"))))\n",
    "\n",
    "\n",
    "data3groups = filter(data, diagnosis5 == \"Case_EXP\" |diagnosis5 == \"Case_WT\" | diagnosis5 == \"Control\")\n",
    "dim(data3groups)\n",
    "\n",
    "data3groups = filter(data3groups, age_at_onset > 1)\n",
    "dim(data3groups)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# open raw data file\n",
    "raw = fread(\"/data/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2020-12-2.GRS/Bin/162variants.Merged.Discovery.raw\", header = T)\n",
    "# open raw data file\n",
    "data3groups$FID = data3groups$FID.x\n",
    "\n",
    "data = merge(raw, data3groups, by = \"FID\")\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "data1 = data %>% group_by(diagnosis2, rs113247976_T) %>% \n",
    "  summarize(mean_AAO = mean(age_at_onset), \n",
    "            SD_AAO = sd(age_at_onset))\n",
    "data1 = as.data.table(data1)\n",
    "print(data1)\n",
    "\n",
    "data2 = data %>% group_by(diagnosis2, rs9901522_T) %>% \n",
    "  summarize(mean_AAO = mean(age_at_onset), \n",
    "            SD_AAO = sd(age_at_onset))\n",
    "data2 = as.data.table(data2)\n",
    "print(data2)\n",
    "\n",
    "\n",
    "    \n",
    "data3 = data %>% group_by(diagnosis2) %>% \n",
    "  summarize(mean_AAO = mean(age_at_onset), \n",
    "            SD_AAO = sd(age_at_onset),\n",
    "            n = n())\n",
    "data3 = as.data.table(data3)\n",
    "print(data3)\n",
    "  \n",
    "data3 = data %>% group_by(diagnosis3) %>% \n",
    "  summarize(mean_AAO = mean(age_at_onset), \n",
    "            SD_AAO = sd(age_at_onset),\n",
    "            n = n())\n",
    "data3 = as.data.table(data3)\n",
    "print(data3)\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "e6bb11da",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /gpfs/gsfs11/users/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2022_01_14.ManuscriptRepo\n",
    "\n",
    "echo \"rs113247976\tC9orf72 status\tmean AAO\tSD AAO\tCounts\n",
    "CC\tcarrier\t57.81\t9.49\t780\n",
    "CT\tcarrier\t54.19\t11.09\t37\n",
    "TT\tcarrier\tNA\tNA\t0\n",
    "CC\tnon-carrier\t60.08\t12.54\t6827\n",
    "CT\tnon-carrier\t60.06\t12.14\t206\n",
    "TT\tnon-carrier\t61.75\t11.76\t4\" > rs113247976.table.tab\n",
    "\n",
    "echo \"rs9901522\tC9orf72 status\tmean AAO\tSD AAO\tCounts\n",
    "CC\tcarrier\t58.02762\t9.310195\t724\n",
    "CT\tcarrier\t54.96629\t11.219352\t89\n",
    "TT\tcarrier\t47.75\t6.396614\t4\n",
    "CC\tnon-carrier\t60.04416\t12.527851\t6159\n",
    "CT\tnon-carrier\t60.31776\t12.487447\t856\n",
    "TT\tnon-carrier\t59.72727\t12.78087\t22\" > rs9901522.table.tab\n",
    "\n",
    "\n",
    "echo \"Model\tVariant\tC9orf72 status\tbeta\tp-value\tF-statistics\tp-value F\n",
    "Additive\trs113247976\tcarrier\t-3.360\t0.038\t1.554\t0.050\n",
    "Dominant\trs113247976\tcarrier\t-3.360\t0.038\t1.554\t0.050\n",
    "Additive\trs9901522_T\tcarrier\t-3.513\t4.05E-04\t1.946\t5.90E-03\n",
    "Dominant\trs9901522_T\tcarrier\t-3.488\t9.76E-04\t1.867\t9.31E-03\n",
    "Additive\trs113247976\tnon-carrier\t0.157\t0.715\t10.210\t< 2.2e-16\n",
    "Dominant\trs113247976\tnon-carrier\t0.125\t0.88547\t10.210\t< 2.2e-16\n",
    "Additive\trs9901522_T\tnon-carrier\t0.146\t0.863\t10.210\t< 2.2e-16\n",
    "Dominant\trs9901522_T\tnon-carrier\t0.1845\t0.67926\t10.210\t< 2.2e-16\" > Dom.Add.Models.tab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "03e313dd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rs113247976</th>\n",
       "      <th>C9orf72 status</th>\n",
       "      <th>mean AAO</th>\n",
       "      <th>SD AAO</th>\n",
       "      <th>Counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CC</td>\n",
       "      <td>carrier</td>\n",
       "      <td>57.81</td>\n",
       "      <td>9.49</td>\n",
       "      <td>780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CT</td>\n",
       "      <td>carrier</td>\n",
       "      <td>54.19</td>\n",
       "      <td>11.09</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TT</td>\n",
       "      <td>carrier</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CC</td>\n",
       "      <td>non-carrier</td>\n",
       "      <td>60.08</td>\n",
       "      <td>12.54</td>\n",
       "      <td>6827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CT</td>\n",
       "      <td>non-carrier</td>\n",
       "      <td>60.06</td>\n",
       "      <td>12.14</td>\n",
       "      <td>206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TT</td>\n",
       "      <td>non-carrier</td>\n",
       "      <td>61.75</td>\n",
       "      <td>11.76</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  rs113247976 C9orf72 status  mean AAO  SD AAO  Counts\n",
       "0          CC        carrier     57.81    9.49     780\n",
       "1          CT        carrier     54.19   11.09      37\n",
       "2          TT        carrier       NaN     NaN       0\n",
       "3          CC    non-carrier     60.08   12.54    6827\n",
       "4          CT    non-carrier     60.06   12.14     206\n",
       "5          TT    non-carrier     61.75   11.76       4"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "res = pd.read_csv(\"/gpfs/gsfs11/users/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2022_01_14.ManuscriptRepo/rs113247976.table.tab\",sep=\"\\t\")\n",
    "res\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "d75119f3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rs9901522</th>\n",
       "      <th>C9orf72 status</th>\n",
       "      <th>mean AAO</th>\n",
       "      <th>SD AAO</th>\n",
       "      <th>Counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CC</td>\n",
       "      <td>carrier</td>\n",
       "      <td>58.02762</td>\n",
       "      <td>9.310195</td>\n",
       "      <td>724</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CT</td>\n",
       "      <td>carrier</td>\n",
       "      <td>54.96629</td>\n",
       "      <td>11.219352</td>\n",
       "      <td>89</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TT</td>\n",
       "      <td>carrier</td>\n",
       "      <td>47.75000</td>\n",
       "      <td>6.396614</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CC</td>\n",
       "      <td>non-carrier</td>\n",
       "      <td>60.04416</td>\n",
       "      <td>12.527851</td>\n",
       "      <td>6159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CT</td>\n",
       "      <td>non-carrier</td>\n",
       "      <td>60.31776</td>\n",
       "      <td>12.487447</td>\n",
       "      <td>856</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TT</td>\n",
       "      <td>non-carrier</td>\n",
       "      <td>59.72727</td>\n",
       "      <td>12.780870</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  rs9901522 C9orf72 status  mean AAO     SD AAO  Counts\n",
       "0        CC        carrier  58.02762   9.310195     724\n",
       "1        CT        carrier  54.96629  11.219352      89\n",
       "2        TT        carrier  47.75000   6.396614       4\n",
       "3        CC    non-carrier  60.04416  12.527851    6159\n",
       "4        CT    non-carrier  60.31776  12.487447     856\n",
       "5        TT    non-carrier  59.72727  12.780870      22"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "res = pd.read_csv(\"/gpfs/gsfs11/users/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2022_01_14.ManuscriptRepo/rs9901522.table.tab\",sep=\"\\t\")\n",
    "res\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "538fd2ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Model</th>\n",
       "      <th>Variant</th>\n",
       "      <th>C9orf72 status</th>\n",
       "      <th>beta</th>\n",
       "      <th>p-value</th>\n",
       "      <th>F-statistics</th>\n",
       "      <th>p-value F</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Additive</td>\n",
       "      <td>rs113247976</td>\n",
       "      <td>carrier</td>\n",
       "      <td>-3.3600</td>\n",
       "      <td>0.038000</td>\n",
       "      <td>1.554</td>\n",
       "      <td>0.050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Dominant</td>\n",
       "      <td>rs113247976</td>\n",
       "      <td>carrier</td>\n",
       "      <td>-3.3600</td>\n",
       "      <td>0.038000</td>\n",
       "      <td>1.554</td>\n",
       "      <td>0.050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Additive</td>\n",
       "      <td>rs9901522_T</td>\n",
       "      <td>carrier</td>\n",
       "      <td>-3.5130</td>\n",
       "      <td>0.000405</td>\n",
       "      <td>1.946</td>\n",
       "      <td>5.90E-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Dominant</td>\n",
       "      <td>rs9901522_T</td>\n",
       "      <td>carrier</td>\n",
       "      <td>-3.4880</td>\n",
       "      <td>0.000976</td>\n",
       "      <td>1.867</td>\n",
       "      <td>9.31E-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Additive</td>\n",
       "      <td>rs113247976</td>\n",
       "      <td>non-carrier</td>\n",
       "      <td>0.1570</td>\n",
       "      <td>0.715000</td>\n",
       "      <td>10.210</td>\n",
       "      <td>&lt; 2.2e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Dominant</td>\n",
       "      <td>rs113247976</td>\n",
       "      <td>non-carrier</td>\n",
       "      <td>0.1250</td>\n",
       "      <td>0.885470</td>\n",
       "      <td>10.210</td>\n",
       "      <td>&lt; 2.2e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Additive</td>\n",
       "      <td>rs9901522_T</td>\n",
       "      <td>non-carrier</td>\n",
       "      <td>0.1460</td>\n",
       "      <td>0.863000</td>\n",
       "      <td>10.210</td>\n",
       "      <td>&lt; 2.2e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Dominant</td>\n",
       "      <td>rs9901522_T</td>\n",
       "      <td>non-carrier</td>\n",
       "      <td>0.1845</td>\n",
       "      <td>0.679260</td>\n",
       "      <td>10.210</td>\n",
       "      <td>&lt; 2.2e-16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Model      Variant C9orf72 status    beta   p-value  F-statistics  \\\n",
       "0  Additive  rs113247976        carrier -3.3600  0.038000         1.554   \n",
       "1  Dominant  rs113247976        carrier -3.3600  0.038000         1.554   \n",
       "2  Additive  rs9901522_T        carrier -3.5130  0.000405         1.946   \n",
       "3  Dominant  rs9901522_T        carrier -3.4880  0.000976         1.867   \n",
       "4  Additive  rs113247976    non-carrier  0.1570  0.715000        10.210   \n",
       "5  Dominant  rs113247976    non-carrier  0.1250  0.885470        10.210   \n",
       "6  Additive  rs9901522_T    non-carrier  0.1460  0.863000        10.210   \n",
       "7  Dominant  rs9901522_T    non-carrier  0.1845  0.679260        10.210   \n",
       "\n",
       "   p-value F  \n",
       "0      0.050  \n",
       "1      0.050  \n",
       "2   5.90E-03  \n",
       "3   9.31E-03  \n",
       "4  < 2.2e-16  \n",
       "5  < 2.2e-16  \n",
       "6  < 2.2e-16  \n",
       "7  < 2.2e-16  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "res = pd.read_csv(\"/gpfs/gsfs11/users/ALS_50k/SaraSaez_ALS/PROJECT2_C9_AAO/2022_01_14.ManuscriptRepo/Dom.Add.Models.tab\",sep=\"\\t\")\n",
    "res\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "706f9467",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "python/3.8",
   "language": "python",
   "name": "py3.8"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": true,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "230.390625px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
